### PR TITLE
feat: default catalog repositories in kommander

### PR DESCRIPTION
**Pull request by takirala**
Context  https://jira.d2iq.com/browse/D2IQ-79950
kommander PR https://github.com/mesosphere/kommander/pull/1329

### CODE REVIEW

File: common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml
Comment (cbuto) [inline common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml:13]: 💭 are we always going to point to main? I thought we initially wanted to tag releases and try to match kommander releases. I’m not sure how we could substitute this for the release version though.

Comment (gracedo) [reply common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml:13]: Do we want to use a substitution variable here and set the version in kommander? (we would use the same version as kommander I guess, if we are keeping those versions coupled)

File: common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml
Comment (gracedo) [inline common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml:15]: for airgapped how would we point this to gitea server?

Comment (takirala) [reply common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml:15]: That is up for discussion, but the CLI (during bootstrap) will do that - https://docs.google.com/document/d/18vzWTmr3O7QNvFxbWAgwusMdT2kQ3ZVFE0YZLxT0Kc4/edit#heading=h.g6xak56mm4x3

Comment (makkes) [reply common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml:15]: Do we need to hold this PR until this is decided for good? I'm afraid this would break any air gap testing otherwise.

**Review by makkes — COMMENTED** [review]
What are the impacts of having these resources unversioned on future upgrades?

File: common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml
Comment (takirala) [reply common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml:15]: This will not break airgapped testing. flux will just fail to reconcile the repo thus behaving as the repo is non existent.
(in our current airgapped ci we do not perform any kuttl tests and only check if kommander is installable and this pr doesnt affect any of that)

**Comment by takirala** [comment]
@mesosphere-mergebot test all

File: common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml
Comment (takirala) [reply common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml:13]: Discussed this offline, decided to go with `branch: main` for now and as part of release process, we will swap this to `tag: v2.2.0`. I ll create a PR to reflect this at https://github.com/mesosphere/kommander/blob/main/CONTRIBUTING.md#publishing-a-new-release once this PR is merged.

**Review by gracedo — APPROVED** [review]

File: common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml
Comment (makkes) [reply common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml:15]: > flux will just fail to reconcile the repo

from my point of view this is a problem and something customers will run into if not tackled. IIUC this plainly does not work in air gapped environments then?

Comment (takirala) [reply common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml]: restructured things a bit and moved the repository definitions to installer file. updated strucuture https://github.com/mesosphere/kommander-cli/pull/335#issue-1113034744 keeps things more obvious and it is upto the user to switch the `github.com/m/dkp-catalog-application` urls to local resources when installting kommander.

**Review by cbuto — APPROVED** [review]

**Review by cwyl02 — APPROVED** [review]

File: common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml
Comment (takirala) [reply common/defaults/project/catalog-repositories/dkp-catalog-applications.yaml]: Update the KEP and will update the release notes to change https://github.com/mesosphere/kommander-cli/pull/353/files#diff-2c390594da79f0b90f7bae6f6f8533240bb60d38c58ec8884caee53ba18aa845R10-R12 as part of release process (structure is TBD, but this will be the place to change)

**Pull request merged**
